libraries/Wire/src/Wire.cpp

```diff
@@ -254,27 +296,55 @@ size_t TwoWire::write(uint8_t data)
   }else{
   // in slave send mode
     // reply to master
-    i2c_slave_write_IT(&_i2c,&data,1);
+    if(i2c_slave_write_IT(&_i2c,&data,1) != I2C_OK) {
+      return 0;
+    }
   }
   return 1;
 }
 
-// must be called in:
-// slave tx event callback
-// or after beginTransmission(address)
+/**
+  * @brief  This function must be called in slave Tx event callback or after
+  *         beginTransmission() and before endTransmission().
+  * @param  pdata: pointer to the buffer data
+  * @param  quantity: number of bytes to write
+  * @retval number of bytes ready to write.
+  */
 size_t TwoWire::write(const uint8_t *data, size_t quantity)
 {
+  size_t nb = 0;
+
   if(transmitting){
   // in master transmitter mode
     for(size_t i = 0; i < quantity; ++i){
-      write(data[i]);
+      nb += write(data[i]);
     }
+    return nb;
   }else{
   // in slave send mode
     // reply to master
-    i2c_slave_write_IT(&_i2c,(uint8_t *)data,quantity);
+    if(i2c_slave_write_IT(&_i2c, (uint8_t *)data, quantity) == I2C_OK) {
+      return quantity;
+    }
   }
-  return quantity;
+  return 0;
+}
+
+/**
+  * @brief  This function must be called after beginTransmission() and before
+  *         endTransmission() only once. This function is only use in master mode.
+  * @param  pdata: pointer to the buffer data
+  * @param  quantity: number of bytes to write
+  * @retval number of bytes read. If > 0 then data are present in pdata buffer.
+  */
+size_t TwoWire::writeBuffer(uint8_t *pdata, size_t quantity)
```

Comment:
Add a warning about the pdata to ensure user do not freed or changed this buffer before endTransmission()

Reply:
Done!